brave/brave-core
# Fixed device duplication when Google Account cookies are deleted

This PR disables upstream's methods
```
SyncServiceImpl::OnAccountsCookieDeletedByUserAction()
SyncServiceImpl::OnAccountsInCookieUpdated()
SyncServiceImpl::OnPrimaryAccountChanged()
```

to prevent re-creating of `Sync Client ID` and device duplication in a Sync chain.


Resolves https://github.com/brave/brave-browser/issues/41615



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
I. Scenario with manual logout from GMail

1. Add devices A and B into the sync chain
2. On device B login into GMail
3. On device B logout from GMail
4. Restart browser on Device B
5. Expected device B is not duplicated at A's list


II. Scenario with `Delete sites data` option
1. Add devices A and B into the sync chain
2. On device B login into GMail
3. On device B go to brave://settings/content/siteData and choose 
```
Delete data sites have saved to your device when you close all windows
Sites will probably work as you expect but won't remember you after you close all Brave windows
```
4. Restart browser on Device B
5. Expected device B is not duplicated at A's list
